Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperations/TestOperation.swift

```diff
@@ -0,0 +1,298 @@
+import MongoSwiftSync
+import Nimble
+import TestsCommon
+
+/// Protocol describing the behavior of a spec test "operation"
+protocol TestOperation: Decodable {
```

Comment:
this protocol was updated to have several different protocol requirements depending on the object rather than a single one for all of them. This allowed us to remove the `guard case let .collection = object` boilerplate in every operation's `execute` body.